applications/nrf5340_audio/src/bluetooth/le_audio_bis_headset.c

```diff
@@ -207,64 +239,73 @@ static void pa_sync_lost_cb(struct bt_audio_broadcast_sink *sink)
-	if (base->subgroup_count > CONFIG_BT_AUDIO_BROADCAST_SNK_STREAM_COUNT) {
-		LOG_WRN("Too many channels in subgroup");
-		return;
-	}
-
-	channel_assignment_get(&channel);
-
-	channel = BIT(channel);
+	LOG_DBG("Received BASE with %zu subgroup(s) from broadcast sink", base->subgroup_count);
 
-	LOG_DBG("Received BASE with %d subgroup(s) from broadcast sink", base->subgroup_count);
+	sync_stream_cnt = 0;
 
 	/* Search each subgroup for the BIS of interest */
 	for (int i = 0; i < base->subgroup_count; i++) {
 		for (int j = 0; j < base->subgroups[i].bis_count; j++) {
 			const uint8_t index = base->subgroups[i].bis_data[j].index;
 
-			LOG_DBG("BIS %d   index = %d", j, index);
+			LOG_DBG("BIS %d   index = %hhu", j, index);
+
+			if (bitrate_check((struct bt_codec *)&base->subgroups[i].codec)) {
+				suitable_stream_found = true;
 
-			/* If this is a BIS of interest then attach to and start a stream */
-			if (index == channel) {
-				if (bitrate_check((struct bt_codec *)&base->subgroups[i].codec)) {
-					base_bis_index_bitfield |= BIT(index);
+				bis_index_bitfields[sync_stream_cnt] = BIT(index);
 
-					streams[i].codec =
-						(struct bt_codec *)&base->subgroups[i].codec;
-					print_codec(streams[i].codec);
+				audio_streams[sync_stream_cnt].codec =
+					(struct bt_codec *)&base->subgroups[i].codec;
+				get_codec_info(audio_streams[sync_stream_cnt].codec,
+					       &audio_codec_info[sync_stream_cnt]);
+				print_codec(&audio_codec_info[sync_stream_cnt]);
 
-					LOG_DBG("Stream %d in subgroup %d from broadcast sink", i,
-						j);
+				LOG_DBG("Stream %d in subgroup %d from broadcast sink",
+					sync_stream_cnt, i);
+
+				sync_stream_cnt += 1;
+				if (sync_stream_cnt >= ARRAY_SIZE(audio_streams)) {
+					break;
 				}
 			}
 		}
+
+		if (sync_stream_cnt >= ARRAY_SIZE(audio_streams)) {
+			break;
+		}
 	}
 
-	if (base_bis_index_bitfield) {
+	/* Set the initial active stream based on the defined channel of the device */
+	channel_assignment_get((enum audio_channel *)&active_stream_index);
+	active_audio_stream.stream = &audio_streams[active_stream_index];
+	active_audio_stream.codec = &audio_codec_info[active_stream_index];
+
+	if (suitable_stream_found) {
 		ret = ctrl_events_le_audio_event_send(LE_AUDIO_EVT_CONFIG_RECEIVED);
 		ERR_CHK(ret);
-		bis_index_bitfield = base_bis_index_bitfield & bis_index_mask;
 
+		LOG_DBG("Channel %s active",
+			((active_stream_index == AUDIO_CH_L) ? CH_L_TAG : CH_R_TAG));
 		LOG_DBG("Waiting for syncable");
 	} else {
-		LOG_WRN("Found no suitable streams");
+		LOG_WRN("Found no suitable stream");
 	}
 }
 
 static void syncable_cb(struct bt_audio_broadcast_sink *sink, bool encrypted)
 {
 	int ret;
 
-	if (streams[0].ep->status.state == BT_AUDIO_EP_STATE_STREAMING || !playing_state) {
+	if (active_audio_stream.stream->ep->status.state == BT_AUDIO_EP_STATE_STREAMING ||
+	    !playing_state) {
+		LOG_DBG("EP streaming or not in playing_state (paused or switching stream)");
```

Comment:
This sentence is weird
```suggestion
		LOG_DBG("Syncable received but either in paused_state or already in a stream");
```